hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -169,6 +170,68 @@
+     * as it may be acceptable for a node to operate at a different compatibility version than its codebase version. This method
+     * may return {@code null} if invoked after the ClusterService is constructed and before the node forms a cluster, either
+     * by joining existing members or becoming master of its standalone cluster if it is the first node on the cluster.
+     * Importantly, this is the time during which a lifecycle event with state
+     * {@link com.hazelcast.core.LifecycleEvent.LifecycleState#STARTING} is triggered.
+     *
+     * For example, consider a cluster comprised of nodes running on {@code hazelcast-3.8.jar}. Each node's codebase version
+     * is 3.8.0 and on startup the cluster version is also 3.8.0. After a while, another node joins, running on
+     * {@code hazelcast-3.9.jar}; this node's codebase version is 3.9.0. If deemed compatible, it is allowed to join the cluster.
+     * At this point, the cluster version is still 3.8.0, the 3.9.0 member should be able to adapt its behaviour to be compatible
+     * with other the 3.8.0 members. Once all 3.8.0 members have been shutdown and replaced by other members on codebase
+     * version 3.9.0, still the cluster version will be 3.8.0. At this point, it is possible to update the cluster version to
+     * 3.9.0, since all cluster members will be compatible with the new cluster version, by invoking
+     * {@link #changeClusterVersion(Version)} or {@link #changeClusterVersion(Version, TransactionOptions)}. Once cluster version
+     * is updated to 3.9.0, further communication among members will take place in 3.9.0 and all new features and functionality
+     * of version 3.9.0 will be available.
+     *
+     * @return the version at which this cluster operates.
+     * @since 3.8
+     */
+    Version getClusterVersion();
+
+    /**
+     * Changes the cluster version transactionally. Internally this method uses the same transaction infrastructure as
+     * {@link #changeClusterState(ClusterState)} and the transaction defaults are the same in this case as well
+     * ({@code TWO_PHASE} transaction with durability 1 by default).
+     *
+     * If the requested cluster version is same as the current one, nothing happens.
+     *
+     * If a node of the cluster is not compatible with the given cluster {@code version}, as implemented by
+     * {@link com.hazelcast.instance.NodeExtension#isNodeVersionCompatibleWith(Version)}, then a
+     * {@link com.hazelcast.internal.cluster.impl.VersionMismatchException} is thrown.
+     *
+     * If an invalid version transition is requested, for example changing to a different major version, an
+     * {@link IllegalArgumentException} is thrown.
+     *
+     * @param version new version of the cluster
+     * @since 3.8
+     */
+    void changeClusterVersion(Version version);
+
+    /**
+     * Changes the cluster version transactionally, with the transaction options provided. Internally this method uses the same
```

Comment:
I see.. but method's utility is already limited, right? At least for now... It cannot set cluster version to any number, it's only for upgrade (and downgrade?). 
What if we don't expose it in public api but only via REST, shell script and management center?